File: app/src/main/java/com/example/achalker/starbuzz/DrinkActivity.java
package com.example.achalker.starbuzz;

import android.media.Image;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.ImageView;
import android.widget.TextView;

public class DrinkActivity extends AppCompatActivity {

    public static final String EXTRA_DRINKED = "drinkId";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_drink);
        int drinkId = (Integer)getIntent().getExtras().get(EXTRA_DRINKED);
        Drink drink = Drink.drinks[drinkId];

        TextView name = (TextView)findViewById(R.id.name);
        name.setText(drink.getName());

        TextView description = (TextView)findViewById(R.id.description);
        description.setText(drink.getDescription());

        ImageView image = (ImageView)findViewById(R.id.photo);
        image.setImageResource(drink.getImageResourceId());
        image.setContentDescription(drink.getName());
    }
}

File: app/src/main/java/com/example/achalker/starbuzz/DrinkActivity.kt
package com.example.achalker.starbuzz

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.widget.ImageView
import android.widget.TextView
import com.example.achalker.starbuzz.model.Drink

class DrinkActivity : AppCompatActivity() {

    companion object {
        private const val EXTRA_DRINK_NAME = "EXTRA_DRINK_NAME"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_drink)

        val drinkName = intent.extras?.getString(EXTRA_DRINK_NAME, "")!!
        val drink = Drink.drinks.find { it.name == drinkName }

        val name = findViewById<TextView>(R.id.name)
        name.text = drink?.name

        val description = findViewById<TextView>(R.id.description)
        description.text = drink?.description

        val image = findViewById<ImageView>(R.id.photo)
        drink?.imageResourceId?.let { image.setImageResource(it) }
        image.contentDescription = drink?.name
    }
}